race-logic.js

```diff
@@ -24,9 +24,14 @@ async function raceHamsters(hamsterNames) {
         // Move each hamster forward
         let stepsForThisTurn = [];
 
-        // Create an array of hamster indices and shuffle it
+        // Create an array of hamster indices
         let hamsterIndices = hamsters.map((_, index) => index);
-        hamsterIndices = hamsterIndices.sort(() => generator.random() - 0.5);
+
+        // Shuffle it using Fisher-Yates (Knuth) Shuffle
```

Comment:
A more robust method for shuffling the array would be to implement the Fisher-Yates (or Knuth) shuffle algorithm. This algorithm ensures each permutation of the array is equally likely, providing a fairer and more unpredictable race outcome.

This snippet replaces the sorting-based shuffle with the Fisher-Yates shuffle right before the loop that moves each hamster. This change ensures that the order in which hamsters move each turn is truly random, making the race outcomes more varied and fair.